### ISSUE TITLE
redundant-returns-doc: "yield" is a type of "return"

### ISSUE THREAD

**Issue by moylop260** [PyCQA/pylint#984]
### Steps to reproduce
1. Using the following code:
```python
def my_method():
    """Docstring...
    :returns: Values
    :rtype: generator
    """
    for i in range(5):
        yield i
```
2. Run `pylint -d all -e redundant-returns-doc --load-plugins=pylint.extensions.docparams FILE`

### Current behaviour
Show error of `redundant-returns-doc`

### Expected behaviour
`yield` should be consider a type of `return`

### pylint --version output
pylint 2.0.0,
astroid 1.5.0
Python 2.7.6 (default, Jun 22 2015, 17:58:13)
[GCC 4.8.2]

**Comment by moylop260** [comment]
What do you think about @AWhetter ?

**Comment by AWhetter** [comment]
This is definitely an issue. Yield sections are valid in Google and Numpy docstrings though so I need to have a bit of a think about how to implement it but it shouldn't be too hard.

As always thank you for the report!

**Comment by moylop260** [comment]
I can help to fix it
Could you assign me?

**Comment by AWhetter** [comment]
I can only assign maintainers but feel free to have a go at implementing this.
What I have so far consists of having some way of asking the `Docstring` class if it's docstring type supports yields, then changing the `check_functiondef_returns` if statement to use this to check to determine if it should use either `utils.returns_something` or a new `utils.yields_something`.

That's the minimum to get this issue resolved, but there's a bigger question about what the extension should report about yield statements. I'll put this in a new issue once I've got a better idea of what will need doing.

**Comment by moylop260** [comment]
Offtopic: What I need have to join to maintainers team?

**Comment by PCManticore** [comment]
Offtopic: answered privately, on mail.

@moylop260 if you want to tackle this, feel free to take a stab at it, if @AWhetter does not mind. ;-)

**Comment by PCManticore** [comment]
Should be fixed now by #989
